src/containers/RecentTrades/Market.tsx

```diff
@@ -49,91 +30,75 @@ const handleHighlightValue = (prevValue: string, curValue: string) => {
     );
 };
 
+const RecentTradesMarket = () => {
+    const { formatMessage } = useIntl();
+    const dispatch = useDispatch();
 
-class RecentTradesMarketContainer extends React.Component<Props> {
-    public componentDidMount() {
-        if (this.props.currentMarket) {
-            this.props.tradesFetch(this.props.currentMarket);
-        }
-    }
-
-    public componentWillReceiveProps(next: Props) {
-        if (next.currentMarket && this.props.currentMarket !== next.currentMarket) {
-            this.props.tradesFetch(next.currentMarket);
-        }
-    }
-
-    public shouldComponentUpdate(nextProps: Props) {
-        return JSON.stringify(nextProps.recentTrades) !== JSON.stringify(this.props.recentTrades);
-    }
+    const currentMarket = useSelector(selectCurrentMarket);
+    const currentPrice = useSelector(selectCurrentPrice);
+    const recentTrades = useSelector(selectRecentTradesOfCurrentMarket);
 
-    public render() {
-        return (
-            <div className="pg-recent-trades__markets">
-                <Table
-                    data={this.getTrades(this.props.recentTrades)}
-                    header={this.getHeaders()}
-                    onSelect={this.handleOnSelect}
-                />
-            </div>
-        );
-    }
-
-    private getHeaders = () => {
+    const headers = React.useMemo(() => {
         return [
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.time' }),
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.amount' }),
-            this.props.intl.formatMessage({ id: 'page.body.trade.header.recentTrades.content.price' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.time' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.amount' }),
+            formatMessage({ id: 'page.body.trade.header.recentTrades.content.price' }),
         ];
-    };
+    }, [formatMessage]);
```

Comment:
`formatMessage` is not changeable, so you dont' have to add it to dependencies

Reply:
I agree,  but we still either have to use `[formatMessage]` or disabling react warning
In both cases the result is the same

What do you think?